### PR TITLE
[Refactor] Move AttentionSpec initialization to Attention module

### LINKED ISSUE #5463: [RFC]: Refactor Attention module
### Motivation.

Source form: https://github.com/vllm-project/vllm-ascend/issues/4629

1. The Attention section has a large amount of code and many branches.
2. The functions related to Cp differ significantly from those of normal Attention, but the coupling is quite severe.
3. There are many masks due to historical reasons.

### Proposed Change.
1. Remove the attention branch -- done  
2. Isolate PCP and DCP  --done
(1) Forward class extraction 
(2) Metadata coupling processing
(3) Builder processing
3. Unify masks, split masks, and delete all other masks (MLA 50%) @weijinqian0 @wujinyuan1@zhenwenqi2024  
(1) unify some masks.
(2) make attention_mask_builder singleton. @LICO1314 @weijinqian0 
https://github.com/vllm-project/vllm-ascend/pull/4870
https://github.com/vllm-project/vllm-ascend/pull/5389 
4. Metadata processing 
(1) model_runner_v1  @wujinyuan1  @zhenwenqi 
(2) Coordinate with model_runner_v2, remove unused and mergeable elements. wait @liurong ready
(3) The parameter list in the build function is aligned with the native VLLM. @JC-ut0 
5. Delete attn_state. @LICO1314 
6. CP attention, abstract parent class. done
(1) mla cp isolate. 
(2) extract common cp module.
https://github.com/vllm-project/vllm-ascend/pull/5490

Image

7. mla_v1 use FIA operator. @LICO1314 1 wait fia ready  https://github.com/vllm-project/vllm-ascend/pull/5704
8. Device adaptor. @weijinqian0   wait A5 ready
https://github.com/vllm-project/vllm-ascend/pull/5735 
9. function called update_attn_params move to attention module. @drslark @LICO1314  ttps://github.com/vllm-project/vllm-ascend/pull/5532  https://github.com/vllm-project/vllm-ascend/pull/6041 
10. AttentionBuilder Inherit from base class in vllm.  @LICO1314   https://github.com/vllm-project/vllm-ascend/pull/5916
11. Add some comments for CommonMetadata and others. https://github.com/vllm-project/vllm-ascend/pull/5789 @LICO1314 
12. move AttentionSpec to Attention @LICO1314  https://github.com/vllm-project/vllm-ascend/pull/5834
### Feedback Period.

_No response_

### CC List.

_No response_

### Any Other Things.

_No response_

### PR DESCRIPTION
**Pull request by LICO1314**
### What this PR does / why we need it?

This PR refactors `get_kv_cache_spec` method to delegate AttentionSpec creation to each attention module's own `get_kv_cache_spec()` method, aligning with the vllm source code structure.

**Changes:**
- Simplify `get_kv_cache_spec` in `model_runner_v1.py` and `cpu_offload_connector.py`
- Remove manual `AttentionType` checks for `Attention` modules
- Delegate spec creation to each attention module's `get_kv_cache_spec` method directly
- Let `MambaBase` layers use their own `get_kv_cache_spec` method
- Keep `use_sparse` hack for `MLAAttention` (DeepSeek DSA mode) as Ascend-specific handling

This change follows RFC #5463 item 12: move AttentionSpec to Attention module.

- Fixes #5463 (item 12)

### Does this PR introduce _any_ user-facing change?

No. This is an internal refactoring that simplifies code structure without changing any external behavior.

### How was this patch tested?

- Syntax validation passed via `python -m py_compile`
- CI tests will verify the changes work correctly with existing test cases
- vLLM version: v0.13.0
- vLLM main: https://github.com/vllm-project/vllm/commit/2f4e6548efec402b913ffddc8726230d9311948d

### CODE REVIEW

File: vllm_ascend/distributed/kv_transfer/kv_pool/cpu_offload/cpu_offload_connector.py [hunk 1]
```diff
@@ -449,80 +447,44 @@ def get_kv_cache_spec(vllm_config: VllmConfig) -> dict[str, KVCacheSpec]:
...
+                    kv_cache_spec[layer_name] = spec
+
+        elif isinstance(attn_module, MambaBase):
+            # Mamba layers also implement get_kv_cache_spec via AttentionLayerBase
+            # Add Ascend-specific check for prefix caching
+            if vllm_config.cache_config.enable_prefix_caching:
+                raise NotImplementedError(
+                    "Prefix caching is not supported for Mamba yet.")
+            if spec := attn_module.get_kv_cache_spec(vllm_config):
+                kv_cache_spec[layer_name] = spec
```
[the remark this answers is not in the document]
Comment (LICO1314) [reply vllm_ascend/distributed/kv_transfer/kv_pool/cpu_offload/cpu_offload_connector.py:501]: This check is already implemented in vllm's MambaBase.get_kv_cache_spec() method (see vllm/model_executor/layers/mamba/abstract.py lines 44-50). By delegating to the module's own method, we inherit this validation automatically.

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -2901,74 +2899,34 @@ def get_kv_cache_spec(self) -> dict[str, KVCacheSpec]:
...
+                        kv_cache_spec[layer_name] = spec
+
+            elif isinstance(attn_module, MambaBase):
+                # Mamba layers also implement get_kv_cache_spec via AttentionLayerBase
+                # Add Ascend-specific check for prefix caching
+                if self.vllm_config.cache_config.enable_prefix_caching:
+                    raise NotImplementedError(
+                        "Prefix caching is not supported for Mamba yet.")
+                if spec := attn_module.get_kv_cache_spec(self.vllm_config):
+                    kv_cache_spec[layer_name] = spec
```
[the remark this answers is not in the document]
Comment (LICO1314) [reply vllm_ascend/worker/model_runner_v1.py:2933]: Yes, this check exists in vllm's MambaBase.get_kv_cache_spec() method (see vllm/model_executor/layers/mamba/abstract.py lines 44-50). The delegated method already validates this constraint, so no additional check is needed in the runner.

**Review by wangxiyuan — APPROVED** [review]

**Comment by wangxiyuan** [comment]
@lidenghui1110 please help reveiw this change. Thanks!
